doc/benchexec-in-container.md

```diff
@@ -0,0 +1,233 @@
+additional tricks are required, especially in an interactive container (e.g.
+one with a shell, where multiple commands can be executed).
+
+The main complication of using BenchExec with Docker is that benchexec needs a
+"clean" cgroup together with enabled controllers. On a standard system, cgroups
+are usually managed by, for example, systemd, from which BenchExec can
+transparently obtain its cgroup. In contrast, in an interactive container, the
+root process usually is just a shell, which is placed in the "root" cgroup of
+the container. Due to the "no internal processes" rule (see the
+[cgroup documentation](https://man.archlinux.org/man/cgroups.7) for more
+information) any (non-root) cgroup can *either* contain processes *or* delegate
+controllers to a child group. Thus, the shell in the root cgroup of the
+container (which is not the root cgroup of the overall system) prevents child
+cgroups with controllers being created. So, we need to move the init process
+(and thus all subsequent ones) into a separte cgroup, which is the purpose of
+the `init.sh` script.
+
+### Adaptation
+
+There are many ways to achieve the required setup and users familiar with
+Docker may choose to adapt the above procedure. There are a few peculiarities
+to be aware of.
+
+The goal is to create a cgroup called `/benchexec` with relevant controllers
+enabled. here, note that it is important that the cgroup has exactly this path,
+as benchexec uses this as a "fallback" on non-systemd setups.
+
+Additionally, you should be aware of the implications of the "no internal
+process"-rule. Effectively, this means that there cannot be any processes in
+the root cgroup of the container (which, notably, is *not* the root cgroup of
+the system). So, for the provided `init.sh` to work, it needs to be directly
+executed (with `ENTRYPOINT [ "/init.sh" ]`). In particular, using, for example,
+`ENTRYPOINT "/init.sh"` would not work, since this creates a shell that then
+runs `init.sh` as a sub-process, meaning that the root cgroup would not be
+empty when we try to enable subtree control.
+
+In case you run into `Device or resource busy` errors, the problem likely is
+that you want to enable subtree control in a non-empty cgroup or, vice versa,
+move a process to cgroup with enabled subtree control, both of which is
+prohibited for cgroups by design. For debugging, it is useful to inspect the
+`cgroup.procs` and `cgroup.subtree_control` of the cgroup in question. Here,
+note that calls like `cat` do also create a separate process.
+
+Chowning the cgroup and only then switching to the created user is only
+required for Podman. In Docker, it is sufficient to add `USER user` to the
+Dockerfile and at the end of `init.sh` simply add `exec "$@"`. This is due to
+the docker image being executed with `--privileged`, which essentially
+disables proper permissions handling on, e.g., the cgroups, and any user in the
+image can modify the cgroups.
+
+<!--
+## Cgroups v1
```

Comment:
I think we should have this visible in the docs. Maybe above the background part?

Reply:
I just don't know how to do it with cgroups v1

Reply:
I think the current text is fine for this case.